### ISSUE TITLE
Potentially implement fix for 'TopologyException: Input geom 1 is invalid: Ring Self-intersection at or near point' error? [enhancement]

### ISSUE THREAD

**Issue by tati-micheletti** [PredictiveEcology/reproducible#134]
reprex
```
library("reproducible")

studyArea <- Cache(prepInputs,
                   url = "https://drive.google.com/open?id=1LUxoY2-pgkCmmNH5goagBp3IMpj6YrdU",
                   destinationPath = tempdir(),
                   userTags = "edeSA",
                   omitArgs = c("destinationPath"))

firePoly2006 <- prepInputs(url = "https://cwfis.cfs.nrcan.gc.ca/downloads/nbac/nbac_2006_r9_20191129.zip", 
                       studyArea = studyArea,
                       destinationPath = tempdir(), 
                       alsoExtract = "similar",
                       archive = "nbac_2006_r9_20191129.zip",
                       targetFile = "nbac_2006_r9_20191129.shp")

```
The error happens because of intersection problems. Now, according to this (https://github.com/r-spatial/sf/issues/347) `sf` might
be able to deal with it. The error happens in line 338 of postProcess.R (https://github.com/PredictiveEcology/reproducible/blob/development/R/postProcess.R#L338). Just wondering if we could implement the `st_make_valid()` approach for such types of error (that happen BEFORE `fixError()`).

**Comment by eliotmcintire** [comment]
Why wouldn't you put this in `fixErrors`, since that is the objective of `fixErrors`?

**Comment by achubaty** [comment]
do note that (from the `sf` issue you linked) `st_make_valid()` is not available on all `sf` installations, so we need to take are implementing its use

**Comment by achubaty** [comment]
the other suggestion for this problem is to buffer with width zero, which we are already doing I believe.

**Comment by achubaty** [comment]
NOTE:

> Functions and methods that require `liblwgeom`, including `st_make_valid` and all spherical or ellipsoidal metrics (area, distances), have since sf 0.5-5 been moved to their own package, `lwgeom`, which is also on CRAN.

 https://r-spatial.github.io/sf/#lwgeom

**Comment by eliotmcintire** [comment]
Fixed with 89e652ef111af7de91a17a613c66312c1b848847 . Issue was already solved inside `fixErrors`, just hadn't been run by the time the `crop` happens. Now it will run `fixErrors` if the `crop` fails

**Comment by tati-micheletti** [comment]
Works like a charm! Thanks! :)
